File: ios/Classes/Transfer/TransferExecutor.m
#import "TransferExecutor.h"

@implementation TransferExecutor {
    FlutterResult flutterResult;
}

- (void) execute: (NSDictionary *) dictionary result: (FlutterResult) result handler: (id<TransferHandler>) handler {
    flutterResult = result;
    
    UIViewController * viewController = [RootViewControllerProvider get];
    [handler start: viewController dictionary: dictionary delegate: self];

}

- (void)p24TransferOnCanceled {
    flutterResult(@{ @"status" : @"cancel"});
}

- (void)p24TransferOnError:(NSString *)errorCode {
    flutterResult(@{ @"status" : @"error", @"payload" : errorCode});
}

- (void)p24TransferOnSuccess {
    flutterResult(@{ @"status" : @"success"});
}

@end

File: ios/Classes/Transfer/TransferExecutor.swift
import P24

internal class TransferExecutor : NSObject, P24TransferDelegate {
    
    private let flutterResult: FlutterResult
    
    internal init(flutterResult: @escaping FlutterResult) {
        self.flutterResult = flutterResult
    }
    
    internal func execute(dictionary: [String: Any], handler: TransferHandler) {
        handler.start(controller:  RootViewControllerProvider.shared, dictionary: dictionary, delegate: self)
    }
    
    internal func p24TransferOnSuccess() {
        flutterResult(["status" : "success"])
    }
    
    internal func p24Transfer(onError errorCode: String!) {
        flutterResult(["status" : "error", "payload" : errorCode])
    }
    
    internal func p24TransferOnCanceled() {
        flutterResult(["status" : "cancel"])
    }
    
}


